core/arch/arm/kernel/secure_partition.c

```diff
@@ -68,6 +85,31 @@ struct sp_session *sp_get_session(uint32_t session_id)
 	return NULL;
 }
 
+TEE_Result sp_get_partitions_info(struct ffa_partition_info *fpi,
+				  size_t *elements)
+{
+	size_t in_elements = *elements;
+	struct ffa_partition_info *in_fpi = fpi;
+	struct sp_session *s = NULL;
+	uint32_t count = 0;
+
+	TAILQ_FOREACH(s, &open_sp_sessions, link) {
+		if (count < in_elements) {
+			spmc_fill_partition_entry(fpi, s->endpoint_id);
+			fpi++;
+		}
+		count++;
+	}
+
+	if (count > in_elements) {
+		memset(in_fpi, 0, in_elements * sizeof(*fpi));
```

Comment:
Honestly I don't think this is really useful (I mean the `memse()`). Here is what GP has to say about `TEE_ERROR_SHORT_BUFFER`:
```
If the output does not fit in the output buffer, then the implementation SHALL update *size with
the required number of bytes and SHALL return TEE_ERROR_SHORT_BUFFER . It is
implementation-dependent whether the output buffer is left untouched or contains part of the
output. In any case, the TA SHOULD consider that its content is undefined after the function
returns.
```
Filling the buffer with zeroes isn't even mentioned.

Reply:
Done